include/openPMD/backend/Attribute.hpp

```diff
@@ -117,6 +118,20 @@ class Attribute
      */
     template <typename U>
     U get() const;
+
+    /** Retrieve a stored specific Attribute and cast if convertible.
+     *  Like Attribute::get<>(), but returns an empty std::optional if no
+     *  conversion is possible instead of throwing an exception.
+     *
+     * @note This performs a static_cast and might introduce precision loss if
+     *       requested. Check dtype explicitly beforehand if needed.
+     *
+     * @tparam  U   Type of the object to be casted to.
+     * @return  Copy of the retrieved object, casted to type U.
+     *          An empty std::optional if no conversion is possible.
+     */
+    template <typename U>
+    std::optional<U> getOptional() const;
```

Comment:
Please add in Python, too (return `py:None` for `std::nullopt`)

Reply:
Python does not use this call at all:
```
        // C++ pass-through API: Getter
        .def(
            "get_attribute",
            [](Attributable &attr, std::string const &key) {
                auto v = attr.getAttribute(key);
                return v.getResource();
                // TODO instead of returning lists, return all arrays (ndim > 0)
                // as numpy arrays?
            })
```

There is no explicit type requesting as `get<T>()` does, in Python you just get the type that there is

Reply:
Yes, I was wondering if we want to add such an option in Python to cast?